### ISSUE TITLE
google_compute_instance_template doesn't support update

### ISSUE THREAD

**Issue by liamjbennett** [hashicorp/terraform#8606]
From this config:

```
resource "google_compute_instance_template" "web_template" {
  name_prefix = "web"
  machine_type   = "g1-small"

  disk {
    device_name  = "sda"
    source_image = "${var.web_source_image}"
    boot         = true
  }

  network_interface {
    network = "default"

    access_config {

    }
  }
}
```

to this config (with out `access_config`):

```
resource "google_compute_instance_template" "web_template" {
  name_prefix = "web"
  machine_type   = "g1-small"

  disk {
    device_name  = "sda"
    source_image = "${var.web_source_image}"
    boot         = true
  }

  network_interface {
    network = "default"
  }
}
```

The following error this thrown:

```
* google_compute_instance_template.web_template: doesn't support update
```

The fact that it doesn't support updating is fine. I would however expect that it would Delete and Create rather than just throw an error about it.

TF version: 0.7.2

**Comment by cblecker** [comment]
I'm working around this by using lifecycle:
```hcl
  lifecycle {
    # Create a new instance before destroying existing
    create_before_destroy = true
  }
```

and then tainting the template. However, I agree this should happen automatically.

**Comment by danawillow** [comment]
Fixed in #11548
